platform/darwin/test/MGLOfflineStorageTests.mm

```diff
@@ -139,6 +139,22 @@ - (void)testAddPackForBounds {
     }];
     [pack requestProgress];
     [self waitForExpectationsWithTimeout:5 handler:nil];
+    
+    XCTAssertEqualObjects(pack.context, context, @"Offline pack context should match the context specified by the application.");
+    NSString *newName = @"🍑 Peach Grove";
+    NSData *newContext = [NSKeyedArchiver archivedDataWithRootObject:@{
+        nameKey: newName,
+    }];
+    
+    XCTestExpectation *contextCompletionHandlerExpectation = [self expectationWithDescription:@"set pack completion context handler"];
+    __weak MGLOfflinePack *weakPack = pack;
+    [pack setContext:newContext completionHandler:^(NSError * _Nullable error) {
+        XCTAssertNil(error);
+        XCTAssertNotNil(weakPack);
+        XCTAssertEqualObjects(weakPack.context, newContext, @"Offline pack context should match the updated context specified by the application.");
```

Comment:
This assertion currently fails every time due to mapbox/mapbox-gl-native#16450. Until that issue is fixed, the call to `-[MGLOfflinePack setContext:completionHandler:]` will appear to have failed silently, even if it did result in a successful change to the database.

The developer could work around this issue by calling `-[MGLOfflineStorage reloadPacks]` in the completion handler, but that has the side effect of invalidating any MGLOfflinePack object the application is currently holding onto.